src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingSearchService.kt

```diff
@@ -1,71 +1,145 @@
 package uk.gov.justice.digital.hmpps.approvedpremisesapi.service
 
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.BookingSearchSortField
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.BookingStatus
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.ServiceName
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.SortDirection
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.SortOrder
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BookingRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BookingSearchResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.BookingSearchResultDto
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.PaginationMetadata
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.validated
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.repository.BookingSearchRepository
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.repository.BookingSearchResult
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.AuthorisableActionResult
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.ValidatableActionResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.getMetadata
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.getPageable
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import kotlin.Comparator
 
 @Service
 class BookingSearchService(
-  private val bookingSearchRepository: BookingSearchRepository,
   private val offenderService: OffenderService,
   private val userService: UserService,
+  private val bookingRepository: BookingRepository,
 ) {
   fun findBookings(
     serviceName: ServiceName,
     status: BookingStatus?,
     sortOrder: SortOrder,
     sortField: BookingSearchSortField,
-  ): AuthorisableActionResult<ValidatableActionResult<List<BookingSearchResult>>> {
+    page: Int?,
+  ): Pair<AuthorisableActionResult.Success<ValidatableActionResult<List<BookingSearchResultDto>>>, PaginationMetadata?> {
```

Comment:
I think given that we only use this method in one place it would be more understandable to strip out the unnecessary wrapping and just return a `Pair<List<BookingSearchResultDto>, PaginationMetadata?>`.

Reply:
done